order-book/src/test/java/com/market/banica/order/book/model/ItemMarketTest.java

```diff
@@ -0,0 +1,85 @@
+package com.market.banica.order.book.model;
+
+import com.market.Origin;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class ItemMarketTest {
+    private static final String EGGS_ITEM_NAME = "eggs";
+    private static final String RICE_ITEM_NAME = "rice";
+    private static final String MEAT_ITEM_NAME = "meat";
+
+    private final ItemMarket itemMarket = new ItemMarket();
+    private final Map<String, TreeSet<Item>> allItems = new ConcurrentHashMap<>();
+
+    @BeforeEach
+    public void setUp() {
+        TreeSet<Item> items = new TreeSet<>();
+        items.add(new Item(1.2, 3, Origin.EUROPE));
+        items.add(new Item(2.2, 1, Origin.EUROPE));
+        items.add(new Item(3.2, 2, Origin.EUROPE));
+        allItems.put(EGGS_ITEM_NAME, items);
+        allItems.put(RICE_ITEM_NAME, new TreeSet<>());
+
+        ReflectionTestUtils.setField(itemMarket, "allItems", allItems);
+    }
+
+    @Test
+    public void getItemSetByNameWithExistingItemNameReturnsTreeSetOfItems() {
+        //Arrange, Act
+        Optional<Set<Item>> itemSetByName = this.itemMarket.getItemSetByName(EGGS_ITEM_NAME);
+        Set<Item> items = itemSetByName.get();
```

Comment:
this like could be in Assert and to have a null check

Reply:
Fixed